lib/oneview-sdk/image-streamer/resource/api300/golden_image.rb

```diff
@@ -10,13 +10,123 @@
+                end
+              end
+            end
+          end
+          true
+        end
+
+        # Adds an golden image resource from the file that is uploaded from a local drive.
+        # Only the .zip format file can be used for upload.
+        # @param [OneviewSDK::ImageStreamer::Client] client The client object for the Image Streamer appliance
+        # @param [String] file_path
+        # @param [Hash] options The
+        # @option data_options [String] :name The name of the Golden Image
+        # @option data_options [String] :description The description of the Golden Image
+        # @param [Integer] timeout The number of seconds to wait for completing the request
+        # @return [Hash] hash with a String
+        def self.add(client, file_path, data_options = {}, timeout = READ_TIMEOUT)
+          data_options = Hash[data_options.map { |k, v| [k.to_s, v] }] # Convert symbols hash keys to string
+          raise NotFound, "ERROR: File '#{file_path}' not found!" unless File.file?(file_path)
+          raise InvalidFormat, 'ERROR: File with extension not supported!' unless ACCEPTED_FORMATS.include? File.extname(file_path)
+          raise IncompleteResource, 'Please set the name of the golden image!' unless data_options['name']
+          raise IncompleteResource, 'Please set the description of the golden image!' unless data_options['description']
+          options = {}
+          options['Content-Type'] = 'multipart/form-data'
+          options['X-Api-Version'] = client.api_version.to_s
+          options['auth'] = client.token
+          options['file'] = File.basename(file_path)
+          url = URI.parse(URI.escape("#{client.url}#{BASE_URI}"))
+
+          File.open(file_path) do |file|
+            req = Net::HTTP::Post::Multipart.new(
+              url.path,
+              { 'file' => UploadIO.new(file, 'application/octet-stream', File.basename(file_path)) }.merge(data_options),
+              options
+            )
+
+            http_request = Net::HTTP.new(url.host, url.port)
+            http_request.use_ssl = true
+            http_request.verify_mode = OpenSSL::SSL::VERIFY_NONE
+            http_request.read_timeout = timeout
+
+            http_request.start do |http|
+              response = http.request(req)
+              return client.response_handler(response)
+            end
+          end
+        end
+
+        # Sets the os volume
+        # @param [OneviewSDK::ImageStreamer::API300::OsVolumes] os_volume
+        def set_os_volume(os_volume)
+          raise IncompleteResource, 'Please set the OS Volume\'s uri attribute!' unless os_volume['uri']
+          set('osVolumeURI', os_volume['uri'])
+        end
+
+        # Sets the build plan
+        # @param [OneviewSDK::ImageStreamer::API300::BuildPlan] build_plan
+        def set_build_plan(build_plan)
+          raise IncompleteResource, 'Please set the Build Plan\'s uri attribute!' unless build_plan['uri']
```

Comment:
In both of these set methods, it probably wouldn't hurt trying a retrieve before raising the uri not set error, similar to [here](https://github.com/HewlettPackard/oneview-sdk-ruby/blob/master/lib/oneview-sdk/resource/api200/enclosure_group.rb#L55). It would be good to have some consistency between resources